lib/shoryuken/extensions/active_job_adapter.rb

```diff
@@ -53,32 +53,25 @@ def calculate_delay(timestamp)
       def message(queue, job, options = {})
         body = job.serialize
 
-        msg = {}
+        attributes = options.delete(:message_attributes) || {}
+
+        msg = {
+          message_body: body,
+          message_attributes: attributes.merge(MESSAGE_ATTRIBUTES)
+        }
```

Comment:
@timbreitkreutz Sorry for taking long to get back on this. 

Is it being used when enqueuing a job? `MyJob.perform_later(params)` 

I do miss the ability to send Shoryuken specific params (for example, FIFO params) when using Active Job is this change adding this support?

Reply:
Sorry was on vacation.  Will try to get an answer soon :)

Reply:
Sadly I don't think there is a way to inject this via `perform_later`--unless I'm missing something.  This PR was actually aimed more at the middleware layer.

Reply:
@phstc I was able to test this locally. The proposed changes do not permit specification of `message_attributes` when using `perform_later` (I've documented that as a separate issue in #646).

Instead, the following can be used to enqueue a message with additional `message_attributes`, without overriding the `shoryuken_class` attribute.

```ruby
class AddJob < ActiveJob::Base
  def perform(a, b)
    puts a + b
  end
end

job = AddJob.new 1, 2
AddJob.queue_adapter.enqueue(job, message_attributes: { 'attr' => { string_value: 'myval', data_type: 'String' } })
```

Everything looks good from a testing perspective, so I'll merge it. Although it doesn't solve #646, it doesn't hinder it and might actually be an important prerequisite.

@timbreitkreutz Thanks so much for your patience on this one!

Reply:
@timbreitkreutz If you could rebase on `master`, I'd appreciate it. Travis should run correctly after pushing up the rebased version. 🤞

Reply:
Thanks @cjlarose !